sdnfv/openNetVM
# [Bug Fix] Properly Terminate Started but not yet Running NFs

This pr cleans up the NF termination cleanup and resolves a STARTED NF termination which has not entered the RUNNING state.


## Summary:
- Check which state a NF is in, determine cleanup based on that
- Add comments for the people who are debugging this in the future

**Usage:**
Tested with running 1 NF core for the manager, f.e.:
`./go.sh 0,1,2 0 0x60`
Run the NF with bad args, causing it to quit before entering the RUNNING state:
```sh
cd basic_monitor
./go.sh 2 -d 1
```
The above should fail, now run basic monitor with
`./go.sh 2`
Before this pr it would fail, after this pr it should run fine.


| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  
| Breaking API changes     |
| Internal API changes     |
| Usability improvements   |  
| Bug fixes                | :+1: 
| New functionality        |
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 



## Merging notes:
 - Dependencies: None  

**TODO before merging :**



## Test Plan:
Try to break this as hard as you can by running different NFs, with valid/invalid args and stopping them whenever your heart desires.


## Review: 
### Review checklist:  

- Sanity checks, assigned to @kevindweb @Arjun-Vijay
    - Run make in `/onvm` and `/examples` directories
    - Test the bug fix

- Code style, assigned to @kevindweb @Arjun-Vijay
    - Run linter
    - Check everything style related

- Code design, assigned to @kevindweb @Arjun-Vijay
    - Check for memory leaks
    - Check that code is reusable
    - Code is clean, functions are concise
    - Verify that edge cases properly handled

- Performance, assigned to @kevindweb @Arjun-Vijay
    - Run Speed Tester NF, report performance.

- Documentation, assigned to @kevindweb @Arjun-Vijay
    - Check if the new changes are well documented, in both code and READMEs
